cpp/src/arrow/type.cc

```diff
@@ -779,22 +778,20 @@ std::string UnionType::ToString() const {
   return s.str();
 }
 
-SparseUnionType::SparseUnionType(std::vector<std::shared_ptr<Field>> fields,
-                                 std::vector<int8_t> type_codes)
+SparseUnionType::SparseUnionType(FieldVector fields, std::vector<int8_t> type_codes)
     : UnionType(fields, type_codes, Type::SPARSE_UNION) {}
 
-Result<std::shared_ptr<DataType>> SparseUnionType::Make(
-    std::vector<std::shared_ptr<Field>> fields, std::vector<int8_t> type_codes) {
+Result<std::shared_ptr<DataType>> SparseUnionType::Make(FieldVector fields,
+                                                        std::vector<int8_t> type_codes) {
   RETURN_NOT_OK(ValidateParameters(fields, type_codes, UnionMode::SPARSE));
   return std::make_shared<SparseUnionType>(fields, type_codes);
```

Comment:
I wonder would `std::make_shared<SparseUnionType>(std::move(fields), type_codes);` helps here?

Reply:
@mapleFU  I think yes and type_codes can be moved too, but it's not related to this issue?

Reply:
Let's just do the change anyway?

Reply:
ok！will do and I will update the description for this pr.